### PR TITLE
[TACHYON-966] Add back resetConnection in BlockMasterSync

### PR DESCRIPTION
**Pull request by uronce-cc**
https://tachyon.atlassian.net/browse/TACHYON-966

### CODE REVIEW

**Comment by uronce-cc** [comment]
@calvinjia PTAL

File: common/src/main/java/tachyon/MasterClientBase.java [hunk 1]
```diff
@@ -178,6 +178,14 @@ public synchronized void close() {
   }
 
   /**
+   * Closes the connection, then query and set current master address.
+   */
+  public synchronized void resetConnection() {
+    close();
```
Comment (calvinjia) [inline common/src/main/java/tachyon/MasterClientBase.java]: Should this be `disconnect()`?

File: servers/src/main/java/tachyon/worker/block/BlockMasterSync.java [hunk 2]
```diff
@@ -140,17 +140,21 @@ public void run() {
...
+      } catch (Exception e) {
         // An error occurred, retry after 1 second or error if heartbeat timeout is reached
-        LOG.error("Failed to receive or execute master heartbeat command.", ioe);
-        // TODO: Add this method in MasterClientBase
-        //mMasterClient.resetConnection();
+        if (cmdFromMaster == null) {
+          LOG.error("Failed to receive master heartbeat command.", e);
+        } else {
+          LOG.error("Failed to receive or execute master heartbeat command: "
+              + cmdFromMaster.toString() + ".", e);
```
Comment (calvinjia) [inline servers/src/main/java/tachyon/worker/block/BlockMasterSync.java]: You can remove the trailing period

**Comment by calvinjia** [comment]
@uronce-cc left a few comments, also could you open a JIRA ticket for this please?

**Comment by uronce-cc** [comment]
@calvinjia Updated.

**Comment by calvinjia** [comment]
LGTM

**Pull request merged**